localstripe/resources.py

```diff
@@ -2651,8 +2639,9 @@ def __init__(self, subscription=None, plan=None, quantity=1,
 
         quantity = try_convert_to_int(quantity)
         try:
-            assert type(subscription) is str
-            assert subscription.startswith('sub_')
+            if subscription is not None:
```

Comment:
If I understand this change, it made possible to have a `SubscriptionItem` without a `Subscription`.
It's surprising, thus may I ask what was the use case?

Reply:
You understand well!

It's tricky, but on special routes, localstripe may have to construct `SubscriptionItem`s without creating a real `Subscription`. For example when doing a simulation:

    GET /v1/invoices/upcoming?customer=cus_n2FydPZCLKMDXk&subscription_items[0][plan]=my=plan&subscription_default_tax_rates[0]=txr_VZ9GmAd2Ka6km0`

Reply:
OK, I see it, thanks!